packages/nuxt/src/app/components/nuxt-island.ts

```diff
@@ -34,7 +34,8 @@ async function loadComponents (source = appBaseURL, paths: NuxtIslandResponse['c
 
   const promises: Array<Promise<void>> = []
 
-  for (const [component, item] of Object.entries(paths)) {
```

Comment:
This is deliberate, in many of these cases, for greater type safety - it avoids needing to use the (unsafe) non-null assertion.

If you have another solution that preserves type safety I'd be very grateful.

Reply:
I figured it was most likely because of this, but there is a clear downside here.

TypeScript has an issue with nullability with for in loops because it inherently infers the keys as string, regardless of the object's type.

But until that's fixed on TypeScript's end, it would be an inevitable side effect.

However, I personally don't think the performance and memory impacts are a proportional tradeoff to type safety in the loop declaration, especially with these stats:

```ts
const largeObject = {}
for (let i=0; i<100000;i++) {
    largeObject[i] = i
}

console.time("array method")
for (const num of Object.keys(largeObject)) {
    // It doesn't matter what happens within the loop as the main slowdown is the keys iterator
}
console.timeEnd("array method")

console.time("obj method")
for (const num in largeObject) {
    // It doesn't matter what happens within the loop as the main slowdown is the keys iterator
}
console.timeEnd("obj method")
```

Performance results:
![image](https://github.com/user-attachments/assets/0464740e-000b-4dbc-9fbe-1ce55db855ac)
![image](https://github.com/user-attachments/assets/666854ba-96c1-485b-8a49-35c7453a3c72)

Memory results:
![image](https://github.com/user-attachments/assets/5d0b698b-ddc7-49ef-9cc2-34338d00ef95)
![image](https://github.com/user-attachments/assets/b2b61d46-0d6a-412b-9ff2-8cfb5a37e686)

It's also worth noting that the values inside the loop are correctly typed, it's just that TS is struggling to verify it exists because of their flawed type inferring logic.

What do you think?